src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -410,6 +420,9 @@ else if (shouldBeOnSameLine(option, details)) {
         else if (shouldBeAloneOnLine(option, details, getLine(details.rcurly.getLineNo() - 1))) {
             violation = MSG_KEY_LINE_ALONE;
         }
+        else if(isRightcurlyFollowedBySemicolonHasLineBreakAfter(option, details)) {
+            violation = MSG_KEY_LINE_BREAK_AFTER;
+        }
```

Comment:
the flow of program when we test for an anonymous inner class actually comes to ``shouldBeAloneOnLine()`` and considers that as violation rather than considering the last else statement that I have written as violation which is ``isRightcurlyFollowedBySemicolonHasLineBreakAfter``. I am debugging this and will fix this.